edi_oca/models/edi_exchange_record.py

```diff
@@ -237,12 +260,35 @@ def action_exchange_process(self):
         self.ensure_one()
         return self.backend_id.exchange_process(self)
 
+    def action_retry(self):
+        if not self.retryable:
+            return False
+        new_state = self._rollback_state_mapping[self.edi_exchange_state]
+        fname = "edi_exchange_state"
+        self[fname] = new_state
+        display_state = self._fields[fname].convert_to_export(self[fname], self)
+        self.message_post(
+            body=_("Action retry: state moved back to '%s'") % display_state
+        )
+        return True
```

Comment:
That will not work if exchange_file_auto_generate is not checked. As this record will not be processed by the cron. So, at the end, we need to add similar button that I proposed (with less logic, obviously) or add the logic here (I think it is better with the buttons)